src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -24,63 +25,73 @@ internal class PathHelper
         /// <returns>Normalized path</returns>
         internal static string Normalize(string path)
         {
-            // Get the full path
-            StringBuffer fullPath = new StringBuffer(path.Length);
+            if (path.Length < PathInternal.MaxShortPath)
+            {
+                Span<char> initialBuffer = stackalloc char[PathInternal.MaxShortPath];
```

Comment:
I didn't think it would cost much- we're not zeroing out anymore, right? Presumably the extra space on the stack isn't important. Any rule of thumb as to what is worth avoiding?

Reply:
Smaller code with fewer special cases is better.